packages/ui-extensions-server-kit/src/ExtensionServerClient/ExtensionServerClient.ts

```diff
@@ -20,14 +21,14 @@ export class ExtensionServerClient implements ExtensionServer.Client {
 
   constructor(options: DeepPartial<ExtensionServer.Options> = {}) {
     this.id = (Math.random() + 1).toString(36).substring(7);
-    this.options = {
+    this.options = getValidatedOptions({
       ...options,
       connection: {
         automaticConnect: true,
         protocols: [],
         ...(options.connection ?? {}),
       },
-    } as ExtensionServer.Options;
+    }) as ExtensionServer.Options;
```

Comment:
Could we do something like:
```ts
this.options = getValidatedOptions(mergeOptions(DEFAULT_OPTIONS, options));
```

Reply:
`mergeOptions` already calls `getValidatedOptions` internally

Reply:
I can't use mergeOptions here cause it complains of missing props on `protocols` - again, I'd rather not fix it in this PR as it's unrelated to the changes